docs/DafnyRef/integration- rust/IntegrationRust.md

```diff
@@ -71,42 +71,114 @@ For maximum flexibility, Dafny follows these rules to encode types in Rust:
 
 |-------------------------------|-----------------------------|
 |  Dafny type                   |   Rust type                 |
 |-------------------------------|-----------------------------|
-| `C`, `C?` (for class, iterator C) | `object<C>`             |
-| (trait) `T`                   | (trait) `object<dyn T>`     |
-| `array<T>`                    | `object<[T]>`               |
-| `array2<T>`                   | `object<array2<T>>`         |
+| `C`, `C?` (for class, iterator C) | `Object<C>`         (*) |
+| (trait) `T`                   | (trait) `Object<dyn T>` (*) |
+| `array<T>`                    | `Object<[T]>`           (*) |
+| `array2<T>`                   | `Object<array2<T>>`  (*)(*) |
+| `c.x` if `c: C`               | `rd!(c).x`              (*) |
+| `c.x := 2` if `c: C`          | `md!(c).x = 2`          (*) |
+| `a[0]` if `a: array<T>`       | `rd!(a)[0]`             (*) |
+| `a[1] := 2` if  `a: array<T>` | `md!(a)[1] = 2`         (*) |
+
+
+## Encoding of classes and arrays with the option --raw-pointers
+The option `--raw-pointers` ensures that classes only have 8 bytes pointers,
+and array only have 8 bytes pointers + information about the length,
+and point to raw memory. Therefore, pointers have copy semantics and
+no call to any clone method needs to be made.
+However, since pointer comparison in Rust normally compares vtables,
+which is brittle, Dafny use its own pointer types which are wrappers:
+```rs
+pub struct Ptr<T: ?Sized>(pub Option<NonNull<UnsafeCell<T>>>);
+```
 
-## With the option --raw-pointers=true
-https://doc.rust-lang.org/book/appendix-07-nightly-rust.html
 |-------------------------------|-----------------------------|
 |  Dafny type                   |   Rust type                 |
 |-------------------------------|-----------------------------|
-| `C`, `C?` (for class, iterator C) | `*mut C`                |
-| (trait) `T`                   | (trait) `*mut dyn T`        |
-| `array<T>`                    | `*mut [T]`                  |
-| `array2<T>`                   | `*mut array2<T>`            |
+| `C`, `C?` (for class, iterator C) | `Ptr<C>`            (*) |
+| (trait) `T`                   | (trait) `Ptr<dyn T>`    (*) |
+| `array<T>`                    | `Ptr<[T]>`              (*) |
+| `array2<T>`                   | `Ptr<array2<T>>`    (*) (*) |
+| `c.x` if `c: C`               | `read!(c).x`            (*) |
+| `c.x := 2` if `c: C`          | `modify!(c).x = 2`      (*) |
+| `a[0]` if `a: array<T>`       | `read!(a)[0]`           (*) |
+| `a[1] := 2` if  `a: array<T>` | `modify!(a)[1] = 2`     (*) |
 
 (*) Defined in the Dafny runtime
 
+# Externs
+
+You can provide additional `*.rs` files to `dafny translate`, `dafny build` and even `dafny run` (via the `--input` option)
+All the elements imported from the Rust files are going to be available from the generated code via the module `_dafny_externs` that imports everything.
```

Comment:
This seems like an internal implementation detail: I don't think we want to encourage Rust extern code to depend on it. We should call out that it is subject to change, or not even mention it at all.